### PR TITLE
Close streams when connection is lost

### LINKED ISSUE #35: Streams should be closed when the connection is lost
If a connection is lost (lost network connectivity to other side, other side's process terminated), the Serve() loop gets an EOF and closes the Connection's closeChan, but the session's streams are not closed. If you are blocked reading from a stream, Read will not return EOF. We should close the streams when the connection is lost.

I'm working on a fix and hope to have a PR ready for review today or early next week.

### PR DESCRIPTION
**Pull request by ncdc**
Fixes #35

### CODE REVIEW

File: spdy_test.go [hunk 1]
```diff
@@ -252,6 +252,91 @@ func TestHalfClose(t *testing.T) {
...
+	tt := []struct {
+		closeReceiver bool
+		closeSender   bool
+	}{
+		{closeReceiver: true, closeSender: false},
+		{closeReceiver: false, closeSender: true},
+		{closeReceiver: false, closeSender: false},
+	}
+	for tix, tc := range tt {
+		listen := "localhost:7743"
```
Comment (dmcgowan) [inline spdy_test.go]: Can you rotate the listen port here.  I am getting listen failures on my machine.

Something like this worked for me `listen := fmt.Sprintf("localhost:774%d", tix)`

Comment (ncdc) [inline spdy_test.go]: Would it make more sense to close the listener?

Comment (dmcgowan) [inline spdy_test.go]: Probably makes sense to do both, sometimes I will still get timing related failures when relying on a listen close.

Comment (ncdc) [inline spdy_test.go]: Just pushed a change (forced update) to close the listener. Can you see if that helps?

Comment (LK4D4) [inline spdy_test.go]: Probably right way to do such things is `localhost:0` and then use `listener.Addr().String()` for `Dial`

Comment (ncdc) [inline spdy_test.go]: Ok, I've added the Sprintf too.

Comment (ncdc) [inline spdy_test.go]: @LK4D4 I was wondering how to do random ports. Thanks for the tip!

**Comment by dmcgowan** [comment]
LGTM, can you add a sign off to the commit and I will merge

**Pull request merged**